cobc/ChangeLog

```diff
@@ -1,4 +1,22 @@
 
+2023-07-05  Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>
+
+	* replace.c: rewrite the code for preprocessing with a two-phase
+	algorithm. The first phase performs COPY REPLACING on the stream
+	of tokens, while the second phase perform REPLACE on the resulting
+	stream of tokens. This rewriting is closer to the COBOL standard
+	and fixes bug #831 partially.
```

Comment:
Hm, why only partially? []

Reply:
#831 actually describes several bugs. One is in listing mode (and is addressed in PR #92 ) and another is in source mode (the one addressed in this PR). I was more concerned by the second part, as it prevents correct code from being compiled, while the first part is only a problem when using the listing mode (we advised our devs to look at the .i files instead of the listing files to exactly know the result of preprocessing).

Reply:
Only because this should be consistent between these files and 4.x should use boolean types ("backported" via gnulib for old compilers).

Reply:
Would https://github.com/OCamlPro/gnucobol/pull/92 solve the listing part already?

Reply:
I haven't found the time to check yet...